runhouse/main.py

```diff
@@ -679,6 +303,207 @@ def cluster_list(
+    except Exception as e:
+        console.print(f"Failed to bring up the cluster: {e}")
+
+
+@cluster_app.command("down")
+def cluster_down(
+    cluster_name: str = typer.Argument(
+        None,
+        help="The name of cluster to terminate.",
+    ),
+    remove_configs: bool = typer.Option(
+        False,
+        "-rm",
+        "--remove-configs",
+        help="Whether to delete cluster config from Den (default: ``False``).",
+    ),
+    remove_all: bool = typer.Option(
+        False,
+        "-a",
+        "--all",
+        help="Whether to terminate all running clusters saved in Den (default: ``False``).",
+    ),
+    confirm_deletion: bool = typer.Option(
+        False, "-y", "--yes", help="Skip confirmation prompt."
+    ),
+):
+    """Terminate cluster if it is not down. No-op if cluster is already down.
+    This only applies to on-demand clusters, and has no effect on self-managed clusters.
+
+    Example:
+
+        ``$ runhouse cluster down /sashab/rh-basic-cpu``
+    """
+    if not confirm_deletion:
+        if cluster_name:
+            proceed = typer.prompt(f"Terminating: {cluster_name}. Proceed? [Y/n]")
+        elif remove_all:
+            proceed = typer.prompt(
+                "Terminating all running clusters saved in Den. Proceed? [Y/n]"
+            )
+        else:
+            console.print("Cannot determine which cluster to terminate, aborting.")
+            raise typer.Exit(1)
+
+        if proceed.lower() in ["n", "no"]:
+            console.print("Aborted!")
+            raise typer.Exit(0)
+
+    if remove_all:
+        running_den_clusters = Cluster.list(status=ClustersListStatus.running).get(
+            "running_clusters"
+        )
+
+        if not running_den_clusters:
+            console.print("No running clusters saved in Den")
+            raise typer.Exit(0)
+
+        terminated_clusters: int = 0
+        for running_cluster in running_den_clusters:
+            try:
+                current_cluster = rh.cluster(
+                    name=f'/{rns_client.username}/{running_cluster.get("Name")}',
+                    dryrun=True,
+                )
+                if isinstance(current_cluster, rh.OnDemandCluster):
+                    current_cluster.teardown_and_delete() if remove_configs else current_cluster.teardown()
+                    terminated_clusters += 1
+                else:
+                    console.print(
+                        f"{current_cluster.rns_address} is not OnDemand Cluster, cannot be terminated."
+                    )
+            except ValueError:
+                continue
+        console.print(f"Successfully terminated {terminated_clusters} clusters.")
+        raise typer.Exit(0)
+
+    current_cluster = get_cluster_or_local(cluster_name)
+
+    try:
+        if isinstance(current_cluster, rh.OnDemandCluster):
+            current_cluster.teardown_and_delete() if remove_configs else current_cluster.teardown()
+        else:
```

Comment:
if it's not an ondemand cluster and `remove_configs` is specified, we can still do that by calling `rns_client.delete_configs(resource= current_cluster)`